src/pages/dataElements/fields/AggregationTypeField.tsx

```diff
@@ -51,19 +53,36 @@ export function AggregationTypeField() {
 
     return (
         <FieldRFF
+            // See https://final-form.org/docs/react-final-form/types/FieldProps#validate
+            // for why this is necessary
+            key={disabled ? 0 : 1}
```

Comment:
I've done this before using `data`-prop instead. It may be less destructive than destroying all the state by resetting it using `key`. Maybe it could work around the "cannot update component" error below as well?

Reply:
I just tried using `data-key` instead of `key` and it doesn't cause the `validate` function to run again. Maybe I'm not doing exactly what you mean?

Reply:
In your current implementation you could just change the validate function to use the valueType from the `allValues` parameter given to validate instead of the outer scope. Then you don't need reset with key.